components/constellation/constellation.rs

```diff
@@ -1231,6 +1231,15 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
                     warn!("Sending reply to get parent info failed ({:?}).", e);
                 }
             }
+            FromScriptMsg::GetChildBrowsingContextId(browsing_context_id, index, sender) => {
+                // We increment here because the 0th element is the parent browsing context itself
+                let result = self.all_descendant_browsing_contexts_iter(browsing_context_id)
+                    .nth(index + 1)
```

Comment:
Hmm, I can't quite parse the following [excerpt from the spec](https://html.spec.whatwg.org/multipage/window-object.html#windowproxy-getownproperty:windowproxy):

> 2. Set value to the WindowProxy object of the indexth document-tree child browsing context of document's browsing context, sorted in the order that their browsing context container elements were most recently inserted into document, the WindowProxy object of the most recently inserted browsing context container's nested browsing context being last.

Initially I believed that this meant the child's child browsing context are ordered right after the child browsing context itself, which corresponds exactly to the ordering we have in `AllDescendantBrowsingContextsIter`, since the grandchild is inserted before the child's next sibling; however, the spec also mentions the _document-tree child browsing context of document's browsing context_, which may mean that we only care about child browsing context only, and (great-to-the-nth) grandchildren browsing contexts do not matter?